toolchain/lex/tokenized_buffer.h

```diff
@@ -260,55 +260,101 @@ class TokenizedBuffer : public Printable<TokenizedBuffer> {
     int indent;
   };
 
-  struct TokenInfo {
-    TokenKind kind;
+  class TokenInfo {
+   public:
+    auto kind() const -> TokenKind { return TokenKind::FromInt(kind_); }
+    auto has_leading_space() const -> bool { return has_leading_space_; }
+
```

Comment:
The class comment is great, but I think it would still be helpful to move `byte_offset` up, and add a comment that collectively documents the payload accessors and mutators (especially their preconditions). That keeps `byte_offset` from getting lost at the bottom, and avoids the need to add a comment to each payload method (which the style guide arguably calls for, but I don't think would be helpful here).